### PR TITLE
Add ATI Label Helpers

### PR DESCRIPTION
**Pull request by dr3**
Contributes to https://github.com/bbc/simorgh/issues/1310

**Overall change:** Add label helpers for ATI

**Code changes:**

- Add generic and article label helpers
- Update window setting logic for testing

These labels can be seen in use here https://github.com/bbc/simorgh/pull/1520 and they are based off of the table here https://github.com/bbc/simorgh/issues/1310 and discussions with Jim

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/PageViewAnalytics/labelHelpers/article.test.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  });
+
+  it('should use null if service is null', () => {
+    const optimoUrn = getPageIdentifier(null, goodData);
+
+    expect(optimoUrn).toEqual('health::news.articles.desiredValue.page');
+  });
+});
+
+describe('getOptimoUrn', () => {
```
Comment (twinlensreflex) [inline src/app/containers/PageViewAnalytics/labelHelpers/article.test.js:46]: this is an option to avoid duplication here

```
const metadataValueChecker = (
  metadataToCheck,
  method,
  goodData,
  goodReturnValue,
  badData,
  badReturnValue,
) => {
  describe(`get${metadataToCheck}`, () => {
    it('should find value in good data', () => {
      const result = method(goodData);

      expect(result).toEqual(goodReturnValue);
    });

    it('should return null in bad data', () => {
      const result = method(badData);

      expect(result).toEqual(badReturnValue);
    });
  });
};

metadataValueChecker(
  'OptimoUrn',
  getOptimoUrn,
  {
    metadata: {
      locators: {
        optimoUrn: 'desired value',
      },
    },
  },
  'desired value',
  {
    metadata: {
      locators: {
        unknownUrn: 'missed value',
      },
    },
  },
  null,
);

metadataValueChecker(
  'Language',
  getLanguage,
  {
    metadata: {
      passport: {
        language: 'desired value',
      },
    },
  },
  'desired value',
  {
    metadata: {
      locators: {
        unknownUrn: 'missed value',
      },
    },
  },
  null,
);

metadataValueChecker(
  'PromoHeadline',
  getPromoHeadline,
  {
    promo: {
      headlines: {
        seoHeadline: 'desired value',
      },
    },
  },
  'desired value',
  {
    promo: {
      headlines: {
        unknown: 'missed value',
      },
    },
  },
  null,
);
```

File: src/app/containers/PageViewAnalytics/labelHelpers/index.test.js [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    });
+
+    it('should find value in good data', () => {
+      expect(func()).toEqual(null);
+    });
+  });
+};
+
+describe('getDestination', () => {
+  const stuff = [
```
Comment (twinlensreflex) [inline src/app/containers/PageViewAnalytics/labelHelpers/index.test.js]:
```suggestion
  const getDestinationTestScenarios = [
```
maybe a name like this would be more suitable?

File: src/app/containers/PageViewAnalytics/labelHelpers/index.test.js [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+  stuff.forEach(({ isUK, env, expected, summary }) => {
+    it(summary, () => {
+      const destination = getDestination(isUK, env);
+      expect(destination).toEqual(expected);
+    });
+  });
+});
+
+describe('getAppType', () => {
+  const stuff = [
```
Comment (twinlensreflex) [inline src/app/containers/PageViewAnalytics/labelHelpers/index.test.js]:
```suggestion
  const getAppTypeScenarios = [
```
and here

File: src/app/containers/PageViewAnalytics/labelHelpers/index.test.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+    const date = new Date(946729425000); // 12:23:45
+    setWindowValue('Date', () => date);
+
+    const ATITime = getCurrentTime();
+
+    expect(ATITime).toEqual('12x23x45');
+  });
+});
+
+describe('getDeviceLanguage', () => {
```
Comment (twinlensreflex) [inline src/app/containers/PageViewAnalytics/labelHelpers/index.test.js:225]: another deduplication suggestion
```
const windowValueChecker = (
  windowValueToCheck,
  method,
  windowValue,
  windowValueString,
  positiveBehaviourDescription,
  positiveValue,
  positiveExpectedValue,
  negativeBehaviourDescription,
  negativeValue,
  negativeExpectedValue,
) => {
  describe(`get${windowValueToCheck}`, () => {
    afterEach(() => {
      resetWindowValue(windowValueString, windowValue);
    });

    returnsNullWhenOffClient(getDeviceLanguage);

    it(`should ${positiveBehaviourDescription}`, () => {
      setWindowValue(windowValueString, positiveValue);

      const result = method();

      expect(result).toEqual(positiveExpectedValue);
    });

    it(`should ${negativeBehaviourDescription}`, () => {
      setWindowValue(windowValueString, negativeValue);

      const result = method();

      expect(result).toEqual(negativeExpectedValue);
    });
  });
};

windowValueChecker(
  'DeviceLanguage',
  getDeviceLanguage,
  window.navigator,
  'navigator',
  'return navigator language',
  { language: 'abc' },
  'abc',
  "return null if langauge isn't set",
  { language: null },
  null,
);

windowValueChecker(
  'Href',
  getHref,
  window.location,
  'location',
  'return location href',
  { href: 'https://href.com' },
  'https://href.com',
  "return null if href isn't set",
  { href: null },
  null,
);
```

**Review by twinlensreflex — CHANGES_REQUESTED** [review]

File: src/app/containers/PageViewAnalytics/labelHelpers/article.test.js (hunk 1 above)
Comment (dr3) [reply src/app/containers/PageViewAnalytics/labelHelpers/article.test.js:46]: Curious of others thoughts @sareh @pjlee11, ^ does reduce ~duplication~ repetition but maybe in exchange for readability? I want to avoid another set of the metadata tests which are pretty hard to follow.

File: src/app/containers/PageViewAnalytics/labelHelpers/index.test.js [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+      summary: 'should return responsive for null',
+    },
+    {
+      platform: undefined,
+      expected: 'responsive',
+      summary: 'should return responsive for undefined',
+    },
+  ];
+
+  stuff.forEach(({ platform, expected, summary }) => {
```
Comment (dr3) [inline src/app/containers/PageViewAnalytics/labelHelpers/index.test.js]:
```suggestion
  getAppTypeScenarios.forEach(({ platform, expected, summary }) => {
```

File: src/app/containers/PageViewAnalytics/labelHelpers/index.test.js [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+    },
+    {
+      isUK: undefined,
+      env: 'live',
+      expected: 598285,
+      summary: 'should return for live uk when isUK is undefined',
+    },
+  ];
+
+  stuff.forEach(({ isUK, env, expected, summary }) => {
```
Comment (dr3) [inline src/app/containers/PageViewAnalytics/labelHelpers/index.test.js]:
```suggestion
  getDestinationTestScenarios.forEach(({ isUK, env, expected, summary }) => {
```

File: src/app/containers/PageViewAnalytics/labelHelpers/article.test.js (hunk 1 above)
Comment (ChrisBAshton) [reply src/app/containers/PageViewAnalytics/labelHelpers/article.test.js:46]: Agreed, I prefer the tests as-is. If it's length we're worried about, I'd sooner delete all the newlines (they just seem so unnecessary in functions this small!)

**Review by oluoluoxenfree — APPROVED** [review]
LGTM 👍

**Review by sareh — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
Nothing available for test here. Happy for this to be merged and tested once the logic is enabled.

**Review by twinlensreflex — APPROVED** [review]
